server/provider/scheduledMiniblocks.go

```diff
@@ -56,13 +64,16 @@ func hasOnlyNormalMiniblocks(block *data.Block) bool {
 
 func removeProcessedMiniblocksOfBlock(block *data.Block) {
 	removeMiniblocksFromBlock(block, func(miniblock *data.MiniBlock) bool {
-		return miniblock.ProcessingType == string(Processed)
+		return miniblock.ProcessingType == dataBlock.Processed.String()
 	})
 }
 
 func removeScheduledMiniblocks(block *data.Block) {
 	removeMiniblocksFromBlock(block, func(miniblock *data.MiniBlock) bool {
-		return miniblock.ProcessingType == string(Scheduled)
+		hasProcessingTypeScheduled := miniblock.ProcessingType == dataBlock.Scheduled.String()
+		hasConstructionStateNotFinal := miniblock.ConstructionState != dataBlock.Final.String()
+
+		return hasProcessingTypeScheduled && hasConstructionStateNotFinal
```

Comment:
might extract a new variable called `shouldRemove` which will be equal to `hasProcessingTypeScheduled && hasConstructionStateNotFinal`

Reply:
Fixed.